File: Backend/BackendTienda/BackendTienda/DTOs/ProductDTOs.cs
namespace BackendTienda.DTOs
{
    #region requests

    public record CreateProductRequest(

        int categoryId,
        string name,
        decimal price,
        int stockQuantity,
        IFormFile image
    );

    public record ReturnProductRequest(string categoryId);
    #endregion

    #region responses
    public record GetProductResponse(

        int productId,
        string categoryId,
        string categoryName,
        string name,
        decimal price,
        int stockQuantity,
        string imageUrl
    );
    #endregion

}

File: Backend/BackendTiendaMigrado/backendtienda/src/main/java/com/backend/backendtienda/dto/ProductDTOs.java
package com.backend.backendtienda.dto;

import org.springframework.web.multipart.MultipartFile;

import java.math.BigDecimal;

public final class ProductDTOs {

    private ProductDTOs() {
    }

    // ===== requests =====
    public record CreateProductRequest(
            Integer categoryId,
            String name,
            BigDecimal price,
            Integer stockQuantity,
            MultipartFile image) {
    }

    public record ReturnProductRequest(String categoryId) {
    }

    // ===== responses =====
    public record GetProductResponse(
            Integer productId,
            String categoryId,
            String categoryName,
            String name,
            BigDecimal price,
            Integer stockQuantity,
            String imageUrl) {
    }
}